packages/suins/sources/sales/pricing_config.move

```diff
@@ -16,17 +16,19 @@ const EPriceNotSet: u64 = 4;
 public struct Range(u64, u64) has copy, store, drop;
 
 /// A struct that holds the length range and the price of a service.
-public struct PricingConfig has copy, store, drop {
-    pricing: VecMap<Range, u64>,
-}
+public struct PricingConfig has store, drop { pricing: VecMap<Range, u64> }
 
-public struct RenewalConfig has copy, store, drop {
-    config: PricingConfig,
-}
+/// A struct that holds the renewal configuration. Exposed from base pricing
+/// module
+/// to allow easy access to the pricing config by external packages.
+public struct RenewalConfig has store, drop { config: PricingConfig }
 
-/// Calculates the price for a given length.
-/// Aborts with EPriceNotSet if the price for the given length is not set.
-public fun calculate_price(config: &PricingConfig, length: u64): u64 {
+/// Calculates the base price for a given length.
+/// - Base price type is abstracted away. We can switch to a different base.
+///  Our core base will become USDC.
+/// - The price is calculated based on the length of the domain name and the
+/// available ranges.
+public fun calculate_base_price(config: &PricingConfig, length: u64): u64 {
```

Comment:
Lines 55-58 we can remove? Range only constructed in `new_range` and asserted

Reply:
The 55-58 check validates that we have unique ranges in ASC order (no overlaps).

So it's like ([3,5], [6,10], [11,22]), without allowing `([3,5], [6,23], [11,22])`